data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3ObjectWorker.java

```diff
@@ -59,24 +67,31 @@ void parseS3Object(final S3ObjectReference s3ObjectPointer) throws IOException {
 
         final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
 
-        try (final ResponseInputStream<GetObjectResponse> object = s3Client.getObject(getObjectRequest)) {
-            codec.parse(object, record -> {
-                try {
-                    bufferAccumulator.add(record);
-                } catch (final Exception e) {
-                    LOG.error("Failed writing S3 objects to buffer.", e);
-                }
-            });
-        } catch (final Exception e) {
-            LOG.error("Error reading from S3 object: s3ObjectReference={}.", s3ObjectPointer, e);
-            s3ObjectsFailedCounter.increment();
-            throw e;
-        }
 
-        try {
-            bufferAccumulator.flush();
-        } catch (final Exception e) {
-            LOG.error("Failed writing S3 objects to buffer.", e);
-        }
+        s3ObjectReadTimer.wrap((Callable<Void>) () -> {
```

Comment:
Take it or leave it:

This method is beginning to do a lot of things. We now have 5 layers of try/catches and inline function calls and over 30 lines. I would have recommend pulling the logic out into smaller functions with more descriptive names but unfortunately the `bufferAccumulator` is not a class variable and is needed at every level including the consumer definition. (I discovered this the hard way trying to provide an example) 

My only other recommendation would be have the codec interface take in the buffer accumulator as a parameter instead. This would allow us to improve some of the readability.

Reply:
Thanks. I'll look into cleaning this part up.